src/org/exist/storage/NativeBroker.java

```diff
@@ -2021,6 +2021,9 @@ public DocumentImpl getResourceById(final int collectionId, final byte resourceT
             //get the resource uri
             final Value key = new CollectionStore.DocumentKey(collectionId, resourceType, documentId);
             final VariableByteInput vbi = collectionsDb.getAsStream(key);
+            if (vbi == null) {
```

Comment:
I am not sure that this change is meant to be in this PR? It doesn't seem to be related to the topic of the PR. Can you explain please?

Reply:
It is necessary in order to avoid try catch around the call to 
```
 doc = context.getBroker().getResourceById(collectionId, resourceType, documentId);
```

Reply:
Hmm... doesn't a `null` here imply that the `collections.dbx` has become corrrupted?